### PR TITLE
Fixes #5637 - Execution of the get attribute on an object should be performed with the same context from which it was called from.

### LINKED ISSUE #5637: Wrong to change 'this' when using super to access method
Executing following code  : 
```
var v=[];
class C {
        m() {return 23};
}
class D extends C {
        f() {return super.boom();}
}
Object.defineProperty(C.prototype, "boom", {get : function() {
        print(this===v);
        return this.m;
}});
print(new D().f.apply(v,[]));
```
Output of ChakraCore : 
```
false
23
```
However, output of other JS engines : 
```
true
Exception: TypeError: super.boom is not a function. (In 'super.boom()', 'super.boom' is undefined)
```

BT group
2018.08.26

### PR DESCRIPTION
**Pull request by wyrichte**
Fixes #5637

### CODE REVIEW

**Comment by sethbrenith** [comment]
nit: you need a `#` character in the phrase `Fixes #5637` for GitHub to automatically link the PR to the issue.

**Comment by sethbrenith** [comment]
LGTM but I'm no expert in this area

File: test/Miscellaneous/KeepContextInSuper.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                    // not throw an error
+                    return function () { }
+                }
+            }
+            class D extends C {
+                // The current context is "a" and not D since apply() and call()
+                // overrided D.
+                g() { super.f(); }
+                h() { super.f = 5; }
+                i() { super.f; }
```
Comment (boingoing) [inline test/Miscellaneous/KeepContextInSuper.js]: Please add some other super call types:
```javascript
let prop = 'f';
let prop2 = 'meth';
class C {
  // ...
  meth() { /* regular method validating 'this' */ }
}
class D extends C {
  // ...
  x() { return super['f']; }
  y() { return super[prop]; }
  z() { super['f'] = 'value'; }
  aa() { super[prop] = 'value'; }
  ab() { return super['meth'](); }
  ac() { return super[prop2](); }
}
```

Comment (wyrichte) [reply test/Miscellaneous/KeepContextInSuper.js]: Thanks for the extra tests!

Tests using a property accessor with bracket notation (`super[prop]`) will result in a `LdElemI_A` opcode which cannot take another property to be used as the context during the execution of the property access - this limitation may require the addition of another opcode (`LdSuperElemI_A`)

Comment (boingoing) [reply test/Miscellaneous/KeepContextInSuper.js]: I am roughly aware of this. Shall we fix it as part of this PR?

Comment (sethbrenith) [reply test/Miscellaneous/KeepContextInSuper.js]: If you do, please also link #4835

File: lib/Runtime/ByteCode/ByteCodeEmitter.cpp [hunk 2]
```diff
@@ -7781,16 +7781,20 @@ void EmitCallTarget(
...
+                pnodeTarget->location, protoLocation, *thisLocation, cacheId, false);
         }
         else
         {
             *thisLocation = pnodeBinTarget->pnode1->location;
+            EmitMethodFld(pnodeBinTarget, protoLocation, propertyId, byteCodeGenerator, funcInfo);
         }
-
-        EmitMethodFld(pnodeBinTarget, protoLocation, propertyId, byteCodeGenerator, funcInfo);
+
```
Comment (dilijev) [inline lib/Runtime/ByteCode/ByteCodeEmitter.cpp]: nit: trim trailing whitespace in this line

File: test/Miscellaneous/KeepContextInSuper.js (hunk 1 above)
Comment (wyrichte) [reply test/Miscellaneous/KeepContextInSuper.js]: This PR will not fix these tests, though they have been added and commented out.

**Pull request merged**